internal/services/sentinel/sentinel_alert_rule_template_data_source.go

```diff
@@ -165,7 +165,8 @@ func dataSourceSentinelAlertRuleTemplateRead(d *pluginsdk.ResourceData, meta int
 		resp, err = getAlertRuleTemplateByName(ctx, client, workspaceID, name)
 	} else {
 		nameToLog = displayName
-		resp, err = getAlertRuleTemplateByDisplayName(ctx, client, workspaceID, displayName)
+		resp, name, err = getAlertRuleTemplateByDisplayName(ctx, client, workspaceID, displayName)
+		id.AlertRuleTemplateName = name
```

Comment:
```suggestion
		resp, name, err = getAlertRuleTemplateByDisplayName(ctx, client, workspaceID, displayName)
		}
		id := parse.NewSentinelAlertRuleTemplateID(subscriptionId, workspaceID.ResourceGroupName, workspaceID.WorkspaceName, name)
```

Reply:
it might be nicer to parse the id after getting the name, rather than just setting it in the id afterwards?